vortex-array/src/array/bool/compute/scalar_at.rs

```diff
@@ -3,15 +3,13 @@ use vortex_scalar::Scalar;
 
 use crate::array::BoolArray;
 use crate::compute::unary::ScalarAtFn;
-use crate::validity::ArrayValidity;
-use crate::ArrayDType;
 
 impl ScalarAtFn for BoolArray {
     fn scalar_at(&self, index: usize) -> VortexResult<Scalar> {
-        if self.is_valid(index) {
-            Ok(self.boolean_buffer().value(index).into())
-        } else {
-            return Ok(Scalar::null(self.dtype().clone()));
-        }
+        Ok(self.scalar_at_unchecked(index))
```

Comment:
will this still return Scalar::null?

Reply:
The top-level function will do that check, if someone tries to access a logically null value they will probably get `0` but its undefined territory IMO